.adr/0003-consensus-ledger-channels.md

```diff
@@ -0,0 +1,143 @@
+
+FollowerChannel:
+  followerReceive(signedProposal) -> process a proposal received from the channel leader & enqueue it
+  SignNextProposal(expectedProposal) -> inspect the proposalQueue, and if item[0] matches the supplied expectation, sign, dequeue, and prepare for sending to the counterparty
+```
+
+**Note**: state channel security guarantees depend on signed channel states. To that end, proposals sent over the wire are labelled with their turn number, and the signature sent with a proposal is a signature on the **resultant channel state** after the proposed `Add` or `Remove` is applied - not on the proposal data itself.
+
+
+### Integration with virtualfunding protocols
+
+A client `X` initializing a new virtual channel via the `virtualfund` package (`./protocols/virtualfund`) is provided access to its ledger channel via a getter.
+
+The virtualfund objective is responsible for crafting the `Add` proposal for its target channel, and
+- in the case of a leader ledger channel, calling `Propose()` with this proposal to generate a message for the counterparty
+- in the case of a follower ledger channel, applying this constructed `Add` as the `expectedProposal` for `SignNextProposal()`
+
+Each of these produces a SignedProposal, which is returned to the engine as a protocol side effect and sent to the ledger counterpaty.
+
+
+For `virtualdefund`, the above applies with a `Remove` proposal in place of the `Add`.
+
+### Integration with directfunding protocols
+
+Existing `directfund` and `directdefund` protocols operate on the `Channel` struct from `package channel`. The `Channel` struct is the canonical view of a channel.
+
+This requires a conversion steps on each end:
+- `directfund.Objective` contains converter method `CreateConsensusChannel()`, which prepares a ledger channel for usage by the virtualfunding protocols after the directfund is complete
+- package `directdefund` contains converter method `CreateChannelFromConsensusChannel(cc)`, which prepares a `Channel` struct from the ledger channel.
+
+This is not ideal (why create one temp data-type only to immediatly convert it for storage?), and is subject to future refactoring. See [this notion doc](https://www.notion.so/statechannels/Channels-ConsensusChannels-Objectives-9701e24e2bc1491a83fae375e4e4c64a#9c41424d07694150bdc6cf373309757a) for further detail.
+
+### Peer Messaging
+
+Because each proposal implies a specific, sequentially numbered state, the sending and receiving of proposals must be reliable in context of a real-world network. Safeguards must exist against both __dropped messages__ and __message reordering__.
+
+#### Redundancy (current approach)
+
+Clients include all unacknowledged proposals with each message. In the context of a healthy network, this might be expected to result in occasional double transmmissions and rare triple transmissions. This expectation is subject to future benchmarking / simulation (TODO).
```

Comment:
```suggestion
Clients include all unacknowledged proposals with each message. In the context of a healthy network, this might be expected to result in occasional double transmissions and rare triple transmissions. This expectation is subject to future benchmarking / simulation (TODO).
```